test-deps workflow fails with empty npm audit results

## Problem

The `test-deps / audit` job in the Test Reusable Workflows CI fails with:

```
##[error]Execution failed: 
##[error]Action failed with error: Claude execution failed: 
```

## Root Cause

The test runs `npm audit` against the repository, but there's no `package-lock.json` in the test fixtures. The `npm audit` step has `continue-on-error: true`, so it passes, but the resulting empty audit data causes the Claude Code Action step to fail.

## Suggested Fix

Either:
1. Add a test fixture `package-lock.json` with known vulnerabilities for testing
2. Skip the Claude analysis step when audit results are empty (check `if: steps.audit-cmd.outputs.results != ''`)
3. Add `continue-on-error: true` to the Claude analysis step in the test workflow

## Context

Discovered during PR #558 review. This is a pre-existing issue unrelated to that PR's changes.

--- a/.github/workflows/reusable-security-deps.yml
+++ b/.github/workflows/reusable-security-deps.yml
@@ -129,6 +129,7 @@ jobs:
 
       - name: Claude Dependency Analysis
         id: analyze
+        if: steps.audit-cmd.outputs.results != ''
         uses: anthropics/claude-code-action@v1
         with:
           claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}

